### PR TITLE
Remove default dev/pts

### PR DESCRIPTION
**Pull request by rhatdan**
This allows us to run buildah without being root

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
```
cat ~/Dockerfile 
FROM fedora
RUN echo hello
```
```
id 
uid=3267(dwalsh) gid=3267(dwalsh) groups=3267(dwalsh),10(wheel) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
```
```
 ./buildah bud --net=host ~/
STEP 1: FROM fedora
Getting image source signatures
Skipping fetch of repeat blob sha256:e71c36a80ba912dd7a5a9f2f2d6136c148afa19bc7d024bd616b74a0bc7a2774
Copying config sha256:cc510acfcd701a409014118d5f417f0022520802a26c650866b8a9594d75f3a7
 2.29 KiB / 2.29 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
STEP 2: RUN echo hello
hello
STEP 3: COMMIT containers-storage:[vfs@/home/dwalsh/.local/share/containers/storage+/run/user/3267/run]@94f057525b74621db1b8b002166be617a73ee11ee77062aaaaaecb3264544264
Getting image source signatures
Skipping fetch of repeat blob sha256:891e1e4ef82ad02a4ea1f030831f942d722c7694c4db64ca3239c8163b811c58
Skipping fetch of repeat blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
Copying config sha256:2f5099516d2ad3e32b69027e94aefb3fc7cb9872c6cbb6776b69216150984647
 1.56 KiB / 1.56 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
--> 94f057525b74621db1b8b002166be617a73ee11ee77062aaaaaecb3264544264
```

**Comment by rhatdan** [comment]
@umohnani8 @TomSweeneyRedHat @nalind @giuseppe PTAL

**Comment by nalind** [comment]
This (essentially dropping "gid=5" from the mount options) shouldn't be necessary if GID 5 is mapped into the container.

**Comment by rhatdan** [comment]
It would be nice to force --net=host when running as non root, until @giuseppe figures out a way to setup network namespaces in rootless mode.  But running buildah this simply as non root is very cool.

File: run.go [hunk 1]
```diff
@@ -938,6 +938,17 @@ func (b *Builder) Run(command []string, options RunOptions) error {
...
+		g.RemoveMount("/dev/pts")
+		devPts := specs.Mount{
+			Destination: "/dev/pts",
+			Type:        "devpts",
+			Source:      "devpts",
+			Options:     []string{"nosuid", "noexec", "newinstance", "ptmxmode=0666", "mode=0620"},
+		}
+		g.AddMount(devPts)
+	}
+
```
Comment (TomSweeneyRedHat) [inline run.go]: LGTM.
It doesn't fix #882 though does it?

**Comment by giuseppe** [comment]
> This (essentially dropping "gid=5" from the mount options) shouldn't be necessary if GID 5 is mapped into the container.

There is an issue in runc where it still complains if it detects when running in an user namespace (rootless mode) and there are not mappings in the OCI file.  This is a workaround for runc

File: run.go (hunk 1 above)
Comment (giuseppe) [reply run.go]: It should fix: https://github.com/projectatomic/libpod/issues/1147

**Comment by rhatdan** [comment]
Ok I will merge and we can revert when runc is fixed.

**Comment by rhatdan** [comment]
@rh-atomic-bot r=giuseppe

**Pull request closed without merging**